src/components/Slide.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useRef, useState } from 'react'
+import styled from 'styled-components';
+import RightDirection from '@/assets/icons/RightDirection';
+import LeftDirection from '@/assets/icons/LeftDirection';
+
+const Slide = ({ movies }) => {
+    const TOTAL_SLIDES = 9;
+
+    const [currentSlide, setCurrentSlide] = useState(0);
+    const slideRef = useRef(null);
+
+    // const [validNextBtn, setValidNextBtn] = useState(true);
+    // const [validPrevBtn, setValidPrevBtn] = useState(true);
+
+    const NextSlide = () => {
+        if (currentSlide >= TOTAL_SLIDES) {
+            setCurrentSlide(0);
+        } else {
+            setCurrentSlide(currentSlide + 1);
+        }
+    };    
+    
+    const PrevSlide = () => {
+        if (currentSlide === 0) {
+            setCurrentSlide(TOTAL_SLIDES); 
+        } else {
+            setCurrentSlide(currentSlide - 1);
+        }
+    };
+    
+    useEffect(() => {
+        slideRef.current.style.transition = 'all 0.5s ease-in-out';
+        slideRef.current.style.transform = `translateX(-${currentSlide}00%)`;
```

Comment:
위에서 slideRef에 `<HTMLDivElement>`로 타입을 지정해준다면 아래와 같이 쓸 수 있어요
```suggestion
       const slide = slideRef.current;
       if(slide) {
            slide.style.transition = 'all 0.5s ease-in-out';
            slide.style.transform = `translateX(-${currentSlide}00%)`;
       }
```

Reply:
와우 깔끔해요ㅠㅠ 감사합니다!!!